src/main/java/net/snowflake/client/core/arrow/tostringhelpers/ArrowArrayStringRepresentationBuilder.java

```diff
@@ -0,0 +1,27 @@
+package net.snowflake.client.core.arrow.tostringhelpers;
+
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.SnowflakeType;
+
+@SnowflakeJdbcInternalApi
+public class ArrowArrayStringRepresentationBuilder extends ArrowStringRepresentationBuilderBase {
+
+  public ArrowArrayStringRepresentationBuilder() {
+    super();
+    this.append('[');
+  }
+
+  public ArrowStringRepresentationBuilderBase appendValue(String value, SnowflakeType valueType) {
+    super.addCommaIfNeeded();
+    if (shouldQuoteValue(valueType)) {
+      return this.appendQuoted(value);
+    }
+    return this.append(value);
+  }
+
+  @Override
+  public String toString() {
+    this.append(']');
```

Comment:
If `toString()` method is used more than once, then we will append the closure everytime. Maybe it would be more feasible to append the postfix character within the returning `toString()` instead of this place, or even move it a step further and implement the logic within the base class using additional fields, e.g. `prefix` and `postfix`.